src/components/molecules/OakModalCenter/OakModalCenter.tsx

```diff
@@ -0,0 +1,257 @@
+   */
+  onClose: () => void;
+  /**
+   * The DOM container to render the modal portal into.
+   */
+  domContainer?: Element;
+  /**
+   * Slot for the footer of the modal
+   */
+  footerSlot?: ReactNode;
+};
+
+const FocusOnBox = styled(FocusOn)`
+  ${oakBoxCss}
+`;
+
+const FadeInFlex = styled(OakFlex)<{ $state: TransitionStatus }>`
+  opacity: ${({ $state }) => {
+    switch ($state) {
+      case "entered":
+      case "entering":
+        return "1";
+      default:
+        return "0";
+    }
+  }};
+  transition: opacity 0s;
+`;
+
+/**
+ * Centered modal dialog with trapped focus, close button, backdrop click, and escape key handling.
+ *
+ * ## Props
+ *
+ * - **isOpen** \-                  If true the modal will be open, if false it will be closed
+ * - **onClose** \-                 Called when the modal is closed via the close button, backdrop click, or escape key
+ * - **children** \-                The content of the modal. Use with `<OakModalCenterBody>` for best results.
+ * - **domContainer** \-            The DOM container to render the modal portal into.
+ * - **disableBackdropClick?** \-   If true, clicking the backdrop will not call onClose
+ * - **disableEscapeKey?** \-       If true, pressing the escape key will not call onClose
+ * - **hideCloseButton?** \-        If true, the close button will be hidden
+ * - **modalFlexProps?** \-         Override HTMLAttributes & OakFlex props for the modal container
+ * - **backdropFlexProps?** \-      Override HTMLAttributes & OakFlex props for the backdrop container
+ * - **footerSlot?** \-             Fixed area at the bottom of the modal, this will remain fixed in view if the content is scrollable
+ */
+export const OakModalCenter = ({
+  children,
+  domContainer = document.body,
+  isOpen,
+  onClose,
+  disableBackdropClick = false,
+  disableEscapeKey = false,
+  hideCloseButton = false,
+  modalFlexProps,
+  backdropFlexProps,
+  footerSlot,
+}: OakModalCenterProps) => {
+  const [scrollBorders, setScrollBorders] = useState({
+    top: false,
+    bottom: false,
+  });
+  const transitionRef = useRef<HTMLDivElement>(null);
+  const scrollBoxRef = useRef<HTMLDivElement>(null);
+
+  const checkForScroll = useCallback(() => {
+    // To show some borders when the content is scrollable
+    const scrollBoxElem = scrollBoxRef.current;
+    if (!scrollBoxElem) {
+      return;
+    }
+
+    setScrollBorders({
+      top: scrollBoxElem.scrollTop > 0,
+      bottom: scrollBoxElem.scrollHeight > scrollBoxElem.clientHeight,
+    });
+  }, []);
+
+  useLayoutEffect(() => {
+    checkForScroll();
+    scrollBoxRef.current?.addEventListener("scroll", checkForScroll);
+    return () => {
+      scrollBoxRef.current?.removeEventListener("scroll", checkForScroll);
+    };
+  }, [isOpen]);
+
+  useEffect(() => {
+    window.addEventListener("resize", checkForScroll);
+    return () => {
+      window.removeEventListener("resize", checkForScroll);
+    };
+  });
```

Comment:
This section is all for determining if there is a scroll area and if the user has scrolled. It adds a bottom border if there is a scroll area and a top border only when the user begins to scroll. It looks a lot better with this.